### ISSUE TITLE
Error checking for updates. MacOs

### ISSUE THREAD

**Issue by larrysharon** [Jackett/Jackett#8182]
**Please use the search bar** at the top of the page and make sure you are not creating an already submitted issue.
Check closed issues as well, because your issue may have already been fixed.

Please read our [Contributing Guidelines](https://github.com/Jackett/Jackett/blob/master/CONTRIBUTING.md) before submitting your issue to ensure a prompt response to your bug.

### Environment

**OS**: MacOs

**.Net Runtime**: Mono

**Mono Version**: 6.8

**Jackett Version**: 0.15.36.0

**Last Working Jackett Version**: 0.15.36.0

**Are you using a proxy or VPN?** [yes/no] no

### Description

When trying to update get errors:

2020-04-13 20:46:12 | Error | Error checking for updates.
-- | -- | --
2020-04-13 20:46:12 | Error | Failed to get the release list: Forbidden

Running latest version of Jackett and latest version of Mono on macos.

### Logged Error Messages

2020-04-13 20:46:12 | Error | Error checking for updates.
-- | -- | --
2020-04-13 20:46:12 | Error | Failed to get the release list: Forbidden

### Screenshots

[Place any screenshots of the issue here if needed]

**Comment by ilike2burnthing** [comment]
Please provide enhanced logs:
https://github.com/Jackett/Jackett/wiki/Troubleshooting#how-to-provide-a-jackett-enhanced-log-for-a-ticket

**Comment by larrysharon** [comment]
[log.txt](https://github.com/Jackett/Jackett/files/4472563/log.txt)

**Comment by ilike2burnthing** [comment]
```
Error checking for updates. Newtonsoft.Json.JsonSerializationException: Cannot deserialize the current JSON object (e.g. {"name":"value"}) into type 'System.Collections.Generic.List`1[Jackett.Common.Models.GitHub.Release]' because the type requires a JSON array (e.g. [1,2,3]) to deserialize correctly.
To fix this error either change the JSON to a JSON array (e.g. [1,2,3]) or change the deserialized type so that it is a normal .NET type (e.g. not a primitive type like integer, not a collection type like an array or List<T>) that can be deserialized from a JSON object. JsonObjectAttribute can also be added to the type to force it to deserialize from a JSON object.
Path 'message', line 1, position 11.
   at Newtonsoft.Json.Serialization.JsonSerializerInternalReader.CreateObject(JsonReader reader, Type objectType, JsonContract contract, JsonProperty member, JsonContainerContract containerContract, JsonProperty containerMember, Object existingValue)
   at Newtonsoft.Json.Serialization.JsonSerializerInternalReader.CreateValueInternal(JsonReader reader, Type objectType, JsonContract contract, JsonProperty member, JsonContainerContract containerContract, JsonProperty containerMember, Object existingValue)
   at Newtonsoft.Json.Serialization.JsonSerializerInternalReader.Deserialize(JsonReader reader, Type objectType, Boolean checkAdditionalContent)
   at Newtonsoft.Json.JsonSerializer.DeserializeInternal(JsonReader reader, Type objectType)
   at Newtonsoft.Json.JsonSerializer.Deserialize(JsonReader reader, Type objectType)
   at Newtonsoft.Json.JsonConvert.DeserializeObject(String value, Type type, JsonSerializerSettings settings)
   at Newtonsoft.Json.JsonConvert.DeserializeObject[T](String value, JsonSerializerSettings settings)
   at Newtonsoft.Json.JsonConvert.DeserializeObject[T](String value)
   at Jackett.Common.Services.UpdateService.CheckForUpdates() in /Users/runner/runners/2.165.2/work/1/s/src/Jackett.Common/Services/UpdateService.cs:line 129
```

@garfield69 @ngosang @cadatoiva @RoloSoze any ideas?

**Comment by ilike2burnthing** [comment]
Same issue from 2016, though it doesn't help at all - https://github.com/Jackett/Jackett/issues/368

**Comment by garfield69** [comment]
You are focusing on the wrong error message.
the `Cannot deserialize` is due to the fact that the Jackett updateService did not get any of the data it was expecting.
The actual error we need to be focused on is
`Failed to get the release list: Forbidden`
which should not have occurred as https://api.github.com/repos/Jackett/Jackett/releases
is available to all without any requirements for authorisation.
So the question is: why would GitHub fail to return the list and respond with a 403?
I don't know.
@larrysharon  from a browser running on the same machine that is running Jackett, please click on this link https://api.github.com/repos/Jackett/Jackett/releases and see if you get given text or an error. Thanks.

**Comment by larrysharon** [comment]
From the machine running jacket and using Firefox the attached screenshot is what I got.

Kindest Regards,

Larry Anthony Sharon
p: 917-385-1044
f:  954-281-5438
e: larrysharon@gmail.com

**Comment by larrysharon** [comment]
Sorry wrong screenshot.

Kindest Regards,

Larry Anthony Sharon
p: 917-385-1044
f:  954-281-5438
e: larrysharon@gmail.com

**Comment by garfield69** [comment]
Yah, nah. screen shots need to be loaded directly via to GitHub via your browser, not as an email.

Further analysis: this in the log you provided
```
2020-04-13 20:58:41.1257 Debug WebClient(HttpWebClient2NetCore): Returning Forbidden => {"message":"API rate limit exceeded for 173.44.55.179. (But here's the good news: Authenticated requests get a higher rate limit. Check out the documentation for more details.)","documentation_url":"https://developer.github.com/v3/#rate-limiting"}
```
it may be the issue will clear if you suspend Jackett's attempts to look for updates for a day
until GitHub removes the blacklist for the API limit abuse.
So access your Jackett dashboard and scroll to the bottom and check the **Disable Auto-Update** box, then scroll up  a bit and click the **Apply Server Settings**.
Then Tomorrow untick and apply, then try the **check for updates** and see if that works.

**Comment by larrysharon** [comment]
Sorry here it is direct through GitHub.

![Screen Shot 2020-04-13 at 10 10 58 PM](https://user-images.githubusercontent.com/6014612/79179193-13af8f00-7dd5-11ea-9a66-9ededcb11ce8.png)

**Comment by garfield69** [comment]
yeah, so you are currently blacklisted, and you need to wait 24 hours before trying again.
So disable the auto-update and turn it on after 24 hours have passed.
let me know how you go.

**Comment by ilike2burnthing** [comment]
Do you have the program Headphones installed by any chance? https://github.com/Jackett/Jackett/issues/3201

**Comment by larrysharon** [comment]
No. Have never used Headphones. Just Sonarr and Radarr.

Kindest Regards,

Larry Anthony Sharon
p: 917-385-1044
f:  954-281-5438
e: larrysharon@gmail.com

**Comment by garfield69** [comment]
IF the GitHub API limit blacklist is still in place tomorrow, then you need to start looking at your apps for any that can be linked to GitHub, because Jackett usually only does an automated update check once daily, and unless you have been going crazy with the check-for-update button on the Jackett dashboard, you should not have been able to exceeded the 60 API requests per hour limit.

Unless, perhaps, there is some kind of Jackett service crash / service auto restart loop going on, which is something else we may need to check for tomorrow.

**Comment by larrysharon** [comment]
I waited more than 24 hours and am still getting the same error.  Fresh log attached.

[log.txt](https://github.com/Jackett/Jackett/files/4481377/log.txt)

**Comment by garfield69** [comment]
ok. so the first thing we can do is to find how when the rate limit is due to expire so open a terminal and paste this command
`curl -i https://api.github.com/repos/Jackett/Jackett/releases`
if you are being blocked I expect to see something like this
```
HTTP/1.1 403 Forbidden
Date: Tue, 20 Aug 2013 14:50:41 GMT
Status: 403 Forbidden
X-RateLimit-Limit: 60
X-RateLimit-Remaining: 0
X-RateLimit-Reset: 1377013266
{
   "message": "API rate limit exceeded for xxx.xxx.xxx.xxx. (But here's the good news: Authenticated requests get a higher rate limit. Check out the documentation for more details.)",
   "documentation_url": "https://developer.github.com/v3/#rate-limiting"
}
```
after an hour, do it again, and provide the both the `Date` and `X-RateLimit-Reset` values from **Both** tests here so that:
- we can see if the value is going down or up
- we can turn that reset into a timestamp and subtract the Date to get the time-to-wait until reset.

the second thing to mention is that the logs show that Jackett tried to fetch the releases list just 3 times, once at 2020-04-13 20:58:40, then at 2020-04-15 09:16:55 and again at 2020-04-15 09:17:17
so clearly it is not Jackett that is infringing.
This implies that you have another app running on your mac that is making calls to the GitHub API.
But I have no idea how you would go about tracking these down at the moment, its beyond the scope of this Jackett support site to spend time fixing other mac app issues.

**Comment by larrysharon** [comment]
This is what I got after running that command:

HTTP/1.1 403 rate limit exceeded
Date: Wed, 15 Apr 2020 20:35:27 GMT
Server: Varnish
Strict-Transport-Security: max-age=31536000; includeSubdomains; preload
X-Content-Type-Options: nosniff
X-Frame-Options: deny
X-XSS-Protection: 1; mode=block
Content-Security-Policy: default-src 'none'; style-src 'unsafe-inline'
Content-Type: application/json
X-Ratelimit-Limit: 60
X-Ratelimit-Remaining: 0
X-Ratelimit-Reset: 1586985832
Content-Length: 248
X-GitHub-Request-Id: C829:022E:21B48E:37A64F:5E97700F

<="" span="">
Also this mac-mini is just a headless unit that runs nothing but Radarr, Sonarr, Jackett, Viscosity VPN, Hazel and Transmission. Nothing else.

Kindest Regards,

Larry Anthony Sharon
p: 917-385-1044
f:  954-281-5438
e: larrysharon@gmail.com

**Comment by cadatoiva** [comment]
The reset is a unix timestamp, evaluates to 04/15/2020 @ 9:23pm (UTC)

That's literally the only option that makes sense out of the list of conversions I tried.

**Comment by cadatoiva** [comment]
So after 1 hour, you should be unblocked according to the timestamp in the returned request

**Comment by larrysharon** [comment]
But it’s been weeks with the same issue...

Kindest Regards,

Larry Anthony Sharon
p: 917-385-1044
f:  954-281-5438
e: larrysharon@gmail.com

**Comment by cadatoiva** [comment]
Of the programs that you listed, only Radarr, Sonarr, and Jackett use github to run updates. That by itself shouldn't be causing this problem. However, I believe that the rate limit is based on your public IP address, which means every computer on the same network shares the rate limit. It could also be that the limit grows with each illegal request, so it's really hard to say.

This isn't an issue we can fix, as our program is behaving as expected. Some other program on your network is causing the issue.

**Comment by cadatoiva** [comment]
You did say that your Mac Mini is using a VPN, if Jackett is being routed through the VPN and you don't pay for a dedicated IP, then that's the likely issue, because everyone else using the same VPN server with their Jackett is eating up the rate limit.

**Comment by larrysharon** [comment]
You were absolutely right. I moved the VPN to a different server and it fixed the problem instantly.

Off topic but do you know why Jackett would suddenly start showing search results not matching the queried movies or tv shows in addition to the queried results? It wasn’t doing that up until yesterday or today.

Kindest Regards,

Larry Anthony Sharon
Chief Technology Officer
American Autism Association
p: 917-385-1044 f: 954-281-5438
w: myautism.org e: lsharon@myautism.org

**Comment by garfield69** [comment]
>Off topic but do you know why Jackett would suddenly start showing search results not matching the queried movies or tv shows in addition to the queried results? It wasn’t doing that up until yesterday or today.

Most likely because the web site that the indexer is talking with has changed their search engine and now returns OR'ed words instead of AND'ed words.
open a new ticket specifying the indexer(s) you are talking about and we can check to see of they need adjusting for web site changes.

**Comment by larrysharon** [comment]
Kindest Regards,

Larry Anthony Sharon
Chief Executive Officer, Green Star Biodiesel LLC

p: 888-799-4099 f: 954-281-5438 m: 917-385-1044
a: 3370 NW 37th Street Miami, FL 33142
w: gstarbio.com e: larry@gstarbio.com


---------- Forwarded message ----------
From: FlowApplication <info@salesforce.com>
Date: Jun 9, 2020, 12:58 PM -0400
To: accounts@gstarbio.com <accounts@gstarbio.com>
Subject: Error Occurred During Flow "Service_Mamangement_Process": The flow tried to update these records: null. Th...

> Error element myRule_5_A1 (FlowRecordUpdate).
> The flow tried to update these records: null. This error occurred: DUPLICATE_VALUE: duplicate value found: Serial_Number__c duplicates value on record with id: a0G0W00001ilEGA. You can look up ExceptionCode values in the SOAP API Developer Guide.
> Flow Details
> Flow API Name: Service_Mamangement_Process
> Type: Record Change Process
> Version: 13
> Status: Active
> Org: Green Star Biodiesel (00Dd0000000gQPi)
> Flow Interview Details
> Interview Label: Service_Mamangement_Process-13_InterviewLabel
> Current User: Eduard Rozenfeld (005d0000001f9u6)
> Start time: 6/9/2020 12:58 PM
> Duration: 0 seconds
> How the Interview Started
> Eduard Rozenfeld (005d0000001f9u6) started the flow interview.
> Some of this flow's variables were set when the interview started.
> myVariable_old = null
> myVariable_current = Service__c (a0G0W00001qnTV1UAM)
> ASSIGNMENT: myVariable_waitStartTimeAssignment
> {!myVariable_waitStartTimeVariable} Equals {!$Flow.CurrentDateTime}
> Result
> {!myVariable_waitStartTimeVariable} = "6/9/2020 12:58 PM"
> DECISION: myDecision
> Skipped this outcome because its conditions weren't met: myRule_1
> Outcome conditions:
> {!formula_myRule_1} (false) Equals true
>
> Default outcome executed.
> DECISION: myDecision2
> Skipped this outcome because its conditions weren't met: myRule_3
> Outcome conditions:
> {!formula_myRule_3} (false) Equals true
>
> Default outcome executed.
> DECISION: myDecision4
> Outcome executed: myRule_5
> Outcome conditions:
> {!formula_myRule_5} (true) Equals true
> UPDATE RECORDS: myRule_5_A1
> Find all Service__c records where:
> Id Equals {!myVariable_current.Id} (a0G0W00001qnTV1UAM)
> Update the records’ field values.
> Serial_Number__c = {!myVariable_current.Account__r.Serial_Number__c} (00405)
> Result
> Failed to update records that meet the filter criteria.
> Error Occurred: The flow tried to update these records: null. This error occurred: DUPLICATE_VALUE: duplicate value found: Serial_Number__c duplicates value on record with id: a0G0W00001ilEGA. You can look up ExceptionCode values in the SOAP API Developer Guide.
> Salesforce Error ID: 1398386982-166662 (-804958435)
